### PR TITLE
 integrating experience fragments support for the spa editor

### PR DESCRIPTION
**Pull request by niekraaijmakers**
Integrate the latest changes in the spa core, to provide experience fragments OOTB, also in the SPA editor.

## Related Issue

https://github.com/adobe/aem-project-archetype/issues/744

## Motivation and Context

Partners are struggeling with the SPA editor + Experience Fragments. It's possible but requires some hacking.
This bootstraps now along with the new SPA core updates provides a clean solution OOTB.

## How Has This Been Tested?

Tested this locally against live AEM instances on:

Angular
React
General (making sure the normal project doesn't break)

## Screenshots (if appropriate):

Screenshot 2021-06-25 at 08 18 24

Screenshot 2021-06-25 at 08 19 18

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/main/resources/META-INF/archetype-post-generate.groovy [hunk 1]
```diff
@@ -239,7 +239,8 @@ def cleanUpFrontendModule(frontendModules, optionFrontendModule, rootPom, rootDi
     // Not generating SPA: Delete SPA-specific files
     if (optionFrontendModule != "angular" && optionFrontendModule != "react") {
         // Delete app component
-        assert new File("$appsFolder/components/structure/spa").deleteDir()
+        assert new File("$appsFolder/components/spa").deleteDir()
```
Comment (vladbailescu) [inline src/main/resources/META-INF/archetype-post-generate.groovy:242]: This looks unrelated to the rest of the changes.

Comment (niekraaijmakers) [reply src/main/resources/META-INF/archetype-post-generate.groovy:242]: Nice find. This is indeed just a bug  I should separate in another PR. il revert it for now.. we can do it later somehow

**Review by vladbailescu — APPROVED** [review]

**Review by sharanyavinod — APPROVED** [review]
